tv_apps/browser/js/awesomescreen.js

```diff
@@ -3214,5 +3202,95 @@ var Awesomescreen = {
         list.addItem.bind(list, detail.listIndex)
       );
     }).bind(this));
+  },
+
+  /**
+   * defaultContentView show.
+   */
+  defaultContentViewShow: function awesomescreen_defaultContentViewShow() {
+    if( Browser.currentInfo ) {
+      if( Browser.currentInfo.url ) {
+        Browser.debug('stopped for TopSites display...');
+        this.defaultContentViewHidden();
+        Browser.switchCursorMode(true);
+        return;
+      }
+    }
+    this.showAwesomescreen();
+    this.defaultContentView.classList.add('show');
+    if(this.isDisplayedTab() || window.FTE){
+       Browser.switchCursorMode(false);
+    }else{
+       Browser.switchCursorMode(true);
+    }
+  },
+
+  /**
+   * defaultContentView hidden.
+   */
+  defaultContentViewHidden: function awesomescreen_defaultContentViewHidden() {
+    if(!this.isDisplayedTab()) {
+      this.hideAwesomescreen();
+    }
+    this.defaultContentView.classList.remove('show');
+    Toolbar.setToolbarMode('normal');
+    this.privateBrowsingBlock.classList.remove('visible');
+  },
+
+  /**
+   * defaultContentView restore.
+   */
+  restoreDefaultContentView:
+    function awesomescreen_restoreDefaultContentView() {
+    var defaultTabOptionEl =
+      this.defaultContentView.querySelector('.tab-option');
+    this.changeTabContent(defaultTabOptionEl);
+
+//IFDEF_FIREFOX_SYNC
+    FirefoxSyncTabList.restoreListView();
+//ENDIF_FIREFOX_SYNC
+  },
+
+  /**
+   * handle tab navigation event.
+   */
+  handleTabNavigationEvent: function awesomescreen_handleTabNavigationEvent() {
+    var tabOptionList = this.defaultContentView.querySelectorAll('.tab-option');
+    if (tabOptionList.length > 1) {
+      let i = 0,
+          tabOptionLength = tabOptionList.length;
+      for (; i < tabOptionLength; i++) {
+        let tabOption = tabOptionList[i];
+        tabOption.addEventListener('mouseup',
+          this.handleTabNavigationMouseup.bind(this));
+      }
+    }
+  },
+
+  /**
+   * handle tab navigation mouseup.
+   */
+  handleTabNavigationMouseup:
+    function awesomescreen_handleTabNavigationMouseup(e) {
+      this.changeTabContent(e.target);
+  },
+
+  /**
+   * change tab content by tabOption.
+   */
+  changeTabContent: function awesomescreen_changeTabContent(tabOptionEl) {
```

Comment:
Since this function does handle any data related manipulations, we can simply use CSS class changes on the ```#default-content-view``` element to reduce the hassle. For example ```#default-content-view.show-topsites``` and ```#default-content-view.show-synced-tabs```.